src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndex.java

```diff
@@ -185,6 +230,7 @@ private static void loadNativeLibrary() {
         }
     }
 
+    private static native boolean createReferenceIndex(String referenceName, String indexPrefix, String algorithmName);
```

Comment:
Oh, and I agree about keeping the C code minimal.  By the way, my pending PR has code that allows errors to be propagated back to Java.  If you wanted to rebase on that PR you could use that code.